sympy/discrete/recurrences.py

```diff
@@ -109,6 +109,12 @@ def linrec(coeffs, init, n):
     else:
         b += [S.Zero]*(k - len(b)) # remaining initial values default to zero
 
+
+    return b[n] if n < k else sum(u*v for u, v in zip(_linrec_coeffs(c, n), b))
+
+def _linrec_coeffs(c, n):
```

Comment:
Perhaps this could be explicitly public (no _) though not necessarily imported in `__init__.py`. The docstring could refer to `linrec`.